src/pyedb/dotnet/edb_core/cell/primitive/primitive.py

```diff
@@ -62,6 +60,14 @@ def __init__(self, pedb, edb_object):
             "rectangle": bondwire_cross_section_type.BondwireRectangle,
         }
 
+    @property
+    def _core_stackup(self):
+        return self._app.stackup
+
+    @property
+    def _core_net(self):
+        return self._app.nets
+
```

Comment:
```suggestion

```

Reply:
replace layer_name setter with 

```
layers = self._pedb.stackup.layers
        if isinstance(val, str) and val in layers.keys():
            layer = layers[val]._edb_layer
            if layer:
                self.primitive_object.SetLayer(layer)
            else:
                raise AttributeError("Layer {} not found.".format(val))
        elif isinstance(val, type(layers[list(layers.keys())[0]])):
            try:
                self.primitive_object.SetLayer(val._edb_layer)
            except:
                raise AttributeError("Failed to assign new layer on primitive.")
        else:
            raise AttributeError("Invalid input value")
```

Reply:
I think solution 1 is more elegant. It doesn't impact on speed and there is only 1 place where the layer name is setup. Furthermore solution 2 requires much more redundant line codes

Reply:
core.nets is not used in Primitive but that class is inherited in many others and there should be a need to use it

Reply:
Oh, yes. Let's keep it.